### PR TITLE
Fully support es6

### PR DESCRIPTION
**Pull request by jeremierobert-coveo**
use the package.json module field

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "name": "coveo-search-ui-extensions",
     "version": "0.0.5",
     "description": "Small generic components to extend the functionality of Coveo's Search UI framework.",
-    "main": "./bin/js/commonjs/CoveoJsSearchExtensions.js",
+    "main": "./bin/commonjs/CoveoJsSearchExtensions.js",
+    "module": "./bin/es6/index.js",
```
Comment (jeremierobert-coveo) [inline package.json]: That is the important change.

Now it will auto switch to the es6 version when using es6 and the commonjs otherwise.

Comment (louis-bompart) [reply package.json]: Nice! 👍

Comment (louis-bompart) [reply package.json]: Link to doc for ref sake ?

Comment (jeremierobert-coveo) [reply package.json]: https://webpack.js.org/configuration/resolve/#resolvemainfields

C'est aussi support par Rollup.js

Comment (louis-bompart) [reply package.json]: Pas fan du manque d'official npm support but I can live that with that tho

File: config/webpack.config.js [hunk 2]
```diff
@@ -6,11 +6,11 @@ module.exports = {
         CoveoJsSearchExtensions: './src/Index'
     },
     output: {
-        path: path.resolve('./bin/js/commonjs'),
+        path: path.resolve('./bin/commonjs'),
         filename: `[name].js`,
         libraryTarget: 'umd',
         library: 'CoveoExtension',
-        publicPath: '/js/commonjs'
+        publicPath: '/commonjs'
```
Comment (mikegron) [inline config/webpack.config.js]: sa vas modifier le path dans le CDN ca mais jimagine que c'est correct vue que sa vas etre une autre version anyway

Comment (jeremierobert-coveo) [reply config/webpack.config.js]: 👍

**Pull request merged**